test/op_shape_test.cpp

```diff
@@ -2007,6 +2007,37 @@ TEST_CASE(test_scalar_nelemnts)
 
 TEST_CASE(test_scatternd)
 {
+    {
+        // good
+        auto dtype = migraphx::shape::float_type;
+        auto itype = migraphx::shape::int64_type;
+        migraphx::shape ds{dtype, {8}};
+        migraphx::shape is{itype, {4, 1}};
+        migraphx::shape us{dtype, {4}};
+        expect_shape(ds, migraphx::make_op("scatternd_none"), ds, is, us);
+    }
+
+    {
+        // good, broadcasted
+        auto dtype = migraphx::shape::float_type;
+        auto itype = migraphx::shape::int64_type;
+        migraphx::shape ds{dtype, {8}};
+        migraphx::shape is{itype, {4, 1}, {4, 0}};
+        migraphx::shape us{dtype, {4}};
+        expect_shape(ds, migraphx::make_op("scatternd_none"), ds, is, us);
+    }
+
+    {
+        // too many inputs
+        auto dtype = migraphx::shape::float_type;
+        auto itype = migraphx::shape::int64_type;
+        migraphx::shape ds{dtype, {8}};
+        migraphx::shape is{itype, {4, 1}};
+        migraphx::shape us{dtype, {4}};
+        migraphx::shape zs{dtype, {4}};
+        throws_shape(migraphx::make_op("scatternd_none"), ds, is, us, zs);
+    }
+
```

Comment:
Split into different test cases. Plan to separate the rest of the tests rather than have sub-blocks.